setup-worktree.sh

```diff
@@ -29,6 +31,56 @@ case "$1" in
 
 		echo "Worktree created:"
 		git worktree list || { echo "Error: Failed to list worktree."; exit 2; }
+
+		# Generate the base kernel by invoking the backport script from the kernel-backport worktree
+		if [ -x "$WORKTREE_KERNEL_SOURCES/backport.sh" ]; then
+			( 
+				cd "$WORKTREE_KERNEL_SOURCES" && \
+				./backport.sh
+			) || { echo "Error: backport.sh failed"; exit 3; }
+			if [ -d "$WORKTREE_KERNEL_SOURCES/kernel" ]; then
+				echo "Generated kernel is available at: $WORKTREE_KERNEL_SOURCES/kernel"
+				if cd "$WORKTREE_KERNEL_SOURCES/kernel"; then
+					exec "${SHELL:-/bin/sh}" -i
+				else
+					echo "Error: failed to move to $WORKTREE_KERNEL_SOURCES/kernel"
+				fi
+			else
+				echo "Warning: expected kernel directory not found in $WORKTREE_KERNEL_SOURCES"
+			fi
+		else
+			echo "Error: $WORKTREE_KERNEL_SOURCES/backport.sh not found or not executable"
+		fi
+		;;
+	kernel-backport-only)
+		git worktree add "$WORKTREE_KERNEL_SOURCES" kernel-backport/main || { echo "Error: Failed to add worktree for backport/main. Run clean-worktree first"; exit 2; }
+		# Run backport.sh in the kernel-backport worktree
+		if [ -x "$WORKTREE_KERNEL_SOURCES/backport.sh" ]; then
+			( 
+				cd "$WORKTREE_KERNEL_SOURCES" && \
+				./backport.sh
+			) || { echo "Error: backport.sh failed"; exit 3; }
+			if [ -d "$WORKTREE_KERNEL_SOURCES/kernel" ]; then
+				echo "Generated kernel is available at: $WORKTREE_KERNEL_SOURCES/kernel"
+				if cd "$WORKTREE_KERNEL_SOURCES/kernel"; then
+					exec "${SHELL:-/bin/sh}" -i
+				else
+					echo "Error: failed to move to $WORKTREE_KERNEL_SOURCES/kernel"
+				fi
+			else
+				echo "Warning: expected kernel directory not found in $WORKTREE_KERNEL_SOURCES"
+			fi
+		else
+			echo "Error: $WORKTREE_KERNEL_SOURCES/backport.sh not found or not executable"
```

Comment:
Now I see that this can be further optimized by moving kernel creation to fucntion and call it from options. this will be modular.